assets/tests/components/lateView.test.tsx

```diff
@@ -175,4 +182,602 @@ describe("LateView", () => {
+        lateVehicle2,
+        missingLogonGhost1,
+        missingLogonGhost2,
+      ],
+    }
+
+    const wrapper = mount(
+      <StateDispatchProvider state={state} dispatch={jest.fn()}>
+        <VehiclesByRouteIdProvider vehiclesByRouteId={vehiclesByRouteId}>
+          <LateView />
+        </VehiclesByRouteIdProvider>
+      </StateDispatchProvider>
+    )
+    act(() => {
+      wrapper
+        .find(
+          '.m-late-view__missing-logons .m-late-view__data-row input[type="checkbox"]'
+        )
+        .first()
+        .simulate("click")
+
+      wrapper
+        .find(
+          '.m-late-view__late-buses .m-late-view__data-row input[type="checkbox"]'
+        )
+        .first()
+        .simulate("click")
+
+      wrapper.find(".m-late-view__hide-popup button").first().simulate("click")
+    })
+
+    expect(wrapper.find(".m-late-view__data-row--selected")).toHaveLength(0)
+
+    expect(wrapper.find(".m-late-view__data-row--unselected")).toHaveLength(2)
+  })
+
+  test("can undo hiding, but only for a limited time", () => {
+    const lateVehicle1 = vehicleFactory.build({
+      routeId: "route",
+      runId: "run1",
+      scheduleAdherenceSecs: 901,
+    })
+    const lateVehicle2 = vehicleFactory.build({
+      routeId: "route",
+      runId: "run2",
+      scheduleAdherenceSecs: 901,
+    })
+    const missingLogonGhost1 = ghostFactory.build({
+      routeId: "route",
+      runId: "run3",
+      scheduledLogonTime: 15301,
+    })
+    const missingLogonGhost2 = ghostFactory.build({
+      routeId: "route",
+      runId: "run4",
+      scheduledLogonTime: 15301,
+    })
+
+    const vehiclesByRouteId = {
+      route: [
+        lateVehicle1,
+        lateVehicle2,
+        missingLogonGhost1,
+        missingLogonGhost2,
+      ],
+    }
+
+    const wrapper = mount(
+      <StateDispatchProvider state={state} dispatch={jest.fn()}>
+        <VehiclesByRouteIdProvider vehiclesByRouteId={vehiclesByRouteId}>
+          <LateView />
+        </VehiclesByRouteIdProvider>
+      </StateDispatchProvider>
+    )
+    act(() => {
+      wrapper
+        .find(
+          '.m-late-view__missing-logons .m-late-view__data-row input[type="checkbox"]'
+        )
+        .first()
+        .simulate("click")
+
+      wrapper
+        .find(
+          '.m-late-view__late-buses .m-late-view__data-row input[type="checkbox"]'
+        )
+        .first()
+        .simulate("click")
+
+      wrapper.find(".m-late-view__hide-popup button").first().simulate("click")
+    })
+
+    expect(wrapper.find(".m-late-view__data-row--selected")).toHaveLength(0)
+
+    expect(wrapper.find(".m-late-view__data-row--unselected")).toHaveLength(2)
+
+    act(() => {
+      wrapper
+        .find(".m-late-view__unhide-popup button")
+        .first()
+        .simulate("click")
+    })
+    expect(wrapper.find(".m-late-view__unhide-popup button")).toHaveLength(0)
+    expect(wrapper.find(".m-late-view__data-row--selected")).toHaveLength(0)
```

Comment:
**suggestion:** Also add some sort of assertion that the rows were in fact unhidden (which, if I'm not mistaken, is basically that there should be 2 unselected rows?).